sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/testhelper/DoFnInvokersTestHelper.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.transforms.reflect.testhelper;
+
+import static org.mockito.Mockito.verify;
+
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.reflect.DoFnInvokersTest;
+import org.mockito.Mockito;
+
+/**
+ * Test helper for {@link DoFnInvokersTest}, which needs to test package-private access to DoFns in
+ * other packages.
+ */
+public class DoFnInvokersTestHelper {
+
+  private static class StaticPrivateDoFn extends DoFn<String, String> {
+    @ProcessElement
+    public void process(ProcessContext c) {}
+  }
+
+  private class InnerPrivateDoFn extends DoFn<String, String> {
+    @ProcessElement
+    public void process(ProcessContext c) {}
+  }
+
+  static class StaticPackagePrivateDoFn extends DoFn<String, String> {
+    @ProcessElement
+    public void process(ProcessContext c) {}
+  }
+
+  class InnerPackagePrivateDoFn extends DoFn<String, String> {
+    @ProcessElement
+    public void process(ProcessContext c) {}
+  }
+
+  public static DoFn<String, String> newStaticPackagePrivateDoFn() {
+    return new StaticPackagePrivateDoFn();
+  }
+
+  public static void verifyStaticPackagePrivateDoFn(DoFn<String, String> fn) {
+    verify((StaticPackagePrivateDoFn) fn).process(Mockito.notNull(DoFn.ProcessContext.class));
+  }
+
+  public DoFn<String, String> newInnerPackagePrivateDoFn() {
+    return new InnerPackagePrivateDoFn();
+  }
+
+  public static void verifyInnerPackagePrivateDoFn(DoFn<String, String> fn) {
+    verify((InnerPackagePrivateDoFn) fn).process(Mockito.notNull(DoFn.ProcessContext.class));
```

Comment:
We were previously testing that this gave us the *right* ProcessContext (or at least the `ProcessContext` returned by the `ExtraContextFactory`). Convince me its no longer necessary? Alternatively, pass the expected context into the verify methods and verify we get the same context as we expect?